host/sgx/sgxquote.c

```diff
@@ -622,3 +710,236 @@ oe_result_t oe_sgx_get_supported_attester_format_ids(
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if ((p_qve_report_info &&
+         qve_report_info_size != sizeof(sgx_ql_qe_report_info_t)) ||
+        (!p_qve_report_info && qve_report_info_size > 0))
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if (quote_size > OE_MAX_UINT32)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    if (TRY_TO_USE_SGX_DCAP_QVL() && _load_sgx_dcap_qvl())
+    {
+        sgx_ql_qve_collateral_t* p_sgx_endorsements = NULL;
+
+        // If user provide OE endorsements buffer, try to use it
+        do
+        {
+            if (p_tcb_info && tcb_info_size > 0 && p_tcb_info_issuer_chain &&
+                tcb_info_size > 0 && p_pck_crl && pck_crl_size > 0 &&
+                p_root_ca_crl && root_ca_crl_size > 0 &&
+                p_pck_crl_issuer_chain && pck_crl_issuer_chain_size > 0 &&
+                p_qe_identity && qe_identity_size > 0 &&
+                p_qe_identity_issuer_chain && qe_identity_issuer_chain_size > 0)
+            {
+                p_sgx_endorsements = (sgx_ql_qve_collateral_t*)oe_malloc(
+                    sizeof(sgx_ql_qve_collateral_t));
+                if (p_sgx_endorsements == NULL)
+                {
+                    OE_RAISE_MSG(
+                        OE_OUT_OF_MEMORY,
+                        "Out of memory while creating SGX QVL endorsements.",
+                        NULL);
+                }
+                memset(p_sgx_endorsements, 0, sizeof(sgx_ql_qve_collateral_t));
+
+                p_sgx_endorsements->version = collateral_version;
+                p_sgx_endorsements->tcb_info = (char*)p_tcb_info;
+                p_sgx_endorsements->tcb_info_size = tcb_info_size;
+                p_sgx_endorsements->tcb_info_issuer_chain =
+                    (char*)p_tcb_info_issuer_chain;
+                p_sgx_endorsements->tcb_info_issuer_chain_size =
+                    tcb_info_issuer_chain_size;
+                p_sgx_endorsements->pck_crl = (char*)p_pck_crl;
+                p_sgx_endorsements->pck_crl_size = pck_crl_size;
+                p_sgx_endorsements->root_ca_crl = (char*)p_root_ca_crl;
+                p_sgx_endorsements->root_ca_crl_size = root_ca_crl_size;
+                p_sgx_endorsements->pck_crl_issuer_chain =
+                    (char*)p_pck_crl_issuer_chain;
+                p_sgx_endorsements->pck_crl_issuer_chain_size =
+                    pck_crl_issuer_chain_size;
+                p_sgx_endorsements->qe_identity = (char*)p_qe_identity;
+                p_sgx_endorsements->qe_identity_size = qe_identity_size;
+                p_sgx_endorsements->qe_identity_issuer_chain =
+                    (char*)p_qe_identity_issuer_chain;
+                p_sgx_endorsements->qe_identity_issuer_chain_size =
+                    qe_identity_issuer_chain_size;
+
+                OE_TRACE_INFO("SGX endorsements from OE SDK are used for quote "
+                              "verification\n");
+            }
+
+        } while (0);
```

Comment:
Thanks for pointing this out, I am using `do{} while(0)` just because I want to bypass logic in some error scenarios. 
I didn't realize `OE_CHECK`, `OE_RAISE` use `goto done`  at first. So it make sense to remove it. I will update code.

Reply:
OK, got it.
Thanks!

Reply:
OK, got it.
Thanks!